### PR TITLE
Revert "Allow updating gemspecs loaded from a Gemfile"

### PR DESCRIPTION
**Pull request by laboone**
### Issue
- https://github.com/dependabot/dependabot-core/issues/7071

### Description
When there is a Gem installed via a symlink, Dependabot removes the symlink.
I have created a repository that reproduces this issue: https://github.com/laboone/monorepo-for-dependabot
In this repository, Dependabot creates a PR that removes the symlink, as shown in https://github.com/laboone/monorepo-for-dependabot/pull/1.

This functionality seems to have been introduced in https://github.com/dependabot/dependabot-core/pull/7051.
This PR is a revert of that change.

### CODE REVIEW

**Comment by jeffwidman** [comment]
👋 The bug report makes sense, but we're in the middle of a deploy freeze for a little bit so I have to hold off shipping this to GitHub's service until that finishes.

Given we're already waiting on☝️, I'd also like @deivid-rodriguez to take a peek at this since he put up the original PR so may have ideas for a way to handle the symlink issue in a way that solves the original issue but also doesn't introduce new problems. He's traveling for the next two weeks, so I'll try to remember to remind him to look at this when he returns, but if I forget please ping in two weeks.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -5,6 +5,12 @@ updates:
     schedule:
       interval: "weekly"
 
+  # No ecosystem folders are watched because they roll up to the omnibus watcher
```
Comment (jurre) [inline .github/dependabot.yml:8]: The commit that introduced this change seems unrelated, can we remove it?

**Comment by deivid-rodriguez** [comment]
I had an initial look. I don't think this is an issue with #7051. I believe that bug fix is valid, just made this issue appeared.

I believe the problem is that we don't have logic to properly fetch files inside symlinked folders from the Github API. We only have logic to fetch directly symlinked files. I'm experimenting with a fix at #7411.

File: .github/dependabot.yml (hunk 1 above)
Comment (deivid-rodriguez) [reply .github/dependabot.yml:8]: Basically the bug fix in #7051 allowed us to no longer have to watch `/omnibus`. So technically, if we revert the bug fix, we should start watching that folder again.

But I don't think we should be reverting that bug fix, so I'm looking into properly fixing this bug, which seems related to symlinked folders.

**Comment by deivid-rodriguez** [comment]
I think #7411 is the right fix, so I'd close this PR in favor of #7411. @laboone would you be able to try out that PR?

**Comment by laboone** [comment]
@deivid-rodriguez 
Thank you for your investigation.
Based on your comment in https://github.com/dependabot/dependabot-core/pull/7411#issuecomment-1580767676, it seems to be working correctly.
I will close this PR:+1:
